src/contexts/AuthContextProvider.tsx

```diff
@@ -0,0 +1,44 @@
+"use client";
+import { ReactNode, useEffect, useState } from "react";
+import { AuthContext } from "./AuthContext";
+import axios from "axios";
+import { setCookie, getCookie, deleteCookie } from "cookies-next";
+
+export function AuthProvider({ children }: { children: ReactNode }) {
+  const [isAuth, setIsAuth] = useState(false);
+
+  useEffect(() => {
+    const token = getCookie("token");
+    if (token) {
+      setIsAuth(true);
+      axios.defaults.headers.common["Authorization"] = `Bearer ${token}`;
+    }
+  }, []);
+
+  const handleSignIn = (token: string) => {
+    setCookie("token", `${token}`);
+    axios.defaults.headers.common["Authorization"] = `Bearer ${token}`;
+    console.log(token);
```

Comment:
lets remove console.log token is a safety data it should be always hidden from users eyes